lib/chai/utils/proxify.js

```diff
@@ -32,6 +39,12 @@ module.exports = function proxify (obj) {
       if (typeof property === 'string' &&
           config.proxyExcludedKeys.indexOf(property) === -1 &&
           !Reflect.has(target, property)) {
+        // Special message for invalid property access of non-chainable methods.
+        if (nonChainableMethodName) {
+          throw Error('Invalid use of Chai method: ' + nonChainableMethodName +
+            '. Invocation required.');
+        }
```

Comment:
@keithamus Agreed it can be better. I struggled with the phrasing quite a bit actually. The reason I ended up with the current phrasing is because many of the alternatives I explored only offer sensible advice part of the time.

For example:

```js
expect(x).to.be.equal.to.true;
// Invalid Chai property: equal.to. Did you mean equal(to)?

expect(x).to.be.a.string.with.length(4);
// Invalid Chai property: string.with. Did you mean string(with)?

expect(x).to.throw.an.error;
// Invalid Chai property: throw.an. Did you mean throw(an)?

expect(x).to.be.within.2.and.4;
// Invalid Chai property: within.2. Did you mean within(2)?
```
I feel uncomfortable with Chai reacting to the user's mistake by suggesting an alternative that's also incorrect, especially in the last case since it demonstrates using `within` with only one argument when it actually requires two.

Could do something like: `Invalid Chai property: equal.pizza. See docs for proper usage of the "equal" assertion`.

Or the high-effort solution with customized messages for each method: `Invalid usage of .within. Correct usage: .within(start, finish)`.